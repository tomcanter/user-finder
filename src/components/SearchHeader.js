// @flow

import React, {
  PropTypes,
} from 'react';
import {
  StyleSheet,
  css,
} from 'aphrodite/no-important';
import Container from '../components/Container';

type Props = {
  children?: any,
};

export default function SearchHeader({children}: Props) {
  return (
    <Container extraStyle={styles.SearchHeader}>
      <div className={css(styles.SearchHeader_inner)}>
        <h1 className={css(styles.SearchHeader_title)}>GitHub User Search</h1>
        <p className={css(styles.SearchHeader_intro)}>
          Browse users and their profiles via <a href="https://developer.github.com/v3/">the GitHub API</a>
        </p>
        <div className={css(styles.SearchHeader_children)}>
          {children}
        </div>
      </div>
    </Container>
  );
}

SearchHeader.propTypes = {
  children: PropTypes.element,
};

const styles = StyleSheet.create({
  SearchHeader: {
    backgroundColor: '#f4f4f4',
    borderBottom: '1px solid #ddd',
  },
  SearchHeader_inner: {
    paddingTop: 15,
    paddingBottom: 15,
  },
  SearchHeader_title: {
    fontSize: 32,
  },
  SearchHeader_intro: {
    marginTop: 5,
  },
  SearchHeader_children: {
    marginTop: 20,
  },
});
