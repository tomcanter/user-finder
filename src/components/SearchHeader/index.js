// @flow

import React, {
  PropTypes,
} from 'react';
import {
  StyleSheet,
  css,
} from 'aphrodite/no-important';

type Props = {
  children?: any,
};

export default function SearchHeader({children}: Props) {
  return (
    <div className={css(styles.SearchHeader)}>
      <div className={css(styles.SearchHeader_text)}>
        <h1 className={css(styles.SearchHeader_title)}>GitHub User Search</h1>
        <p className={css(styles.SearchHeader_desc)}>
          Browse users and their profiles via <a href="https://developer.github.com/v3/">the GitHub API</a>
        </p>
      </div>
      <div className={css(styles.SearchHeader_children)}>
        {children}
      </div>
    </div>
  );
}

SearchHeader.propTypes = {
  children: PropTypes.element,
};

const BREAKPOINT = '@media (min-width: 600px)';

const styles = StyleSheet.create({
  SearchHeader: {
    paddingTop: 15,
    paddingBottom: 15,

    [BREAKPOINT]: {
      display: 'flex',
      justifyContent: 'space-between',
      alignItems: 'center',
    },

  },

  SearchHeader_text: {
    flex: 2,
  },

  SearchHeader_title: {
    fontSize: 28,
  },

  SearchHeader_desc: {
    marginTop: 5,
    fontSize: 14,
  },

  SearchHeader_children: {
    marginTop: 20,
    flex: 1,

    [BREAKPOINT]: {
      margin: 0,
    },
  },
});
