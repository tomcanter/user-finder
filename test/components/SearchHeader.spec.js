import React from 'react';
import renderer from 'react-test-renderer';
import {
  StyleSheetTestUtils,
} from 'aphrodite';

import SearchHeader from '../../src/components/SearchHeader';

describe('Component: SearchHeader', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render', () => {
    const component = renderer.create(
      <SearchHeader>
        <div>inner component</div>
      </SearchHeader>
    );
    expect(component).toMatchSnapshot();
  });


});
