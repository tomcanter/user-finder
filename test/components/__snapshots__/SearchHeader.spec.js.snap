exports[`Component: SearchHeader should render 1`] = `
<div
  className="SearchHeader_qw8dio">
  <div
    className="Container_inner_3jewic">
    <div
      className="SearchHeader_inner_1u3xhva">
      <h1
        className="SearchHeader_title_1cswrtu">
        GitHub User Search
      </h1>
      <p
        className="SearchHeader_intro_h76sf7">
        Browse users and their profiles via 
        <a
          href="https://developer.github.com/v3/">
          the GitHub API
        </a>
      </p>
      <div
        className="SearchHeader_children_w48k6q">
        <div>
          inner component
        </div>
      </div>
    </div>
  </div>
</div>
`;
